Source/Csla.Shared/Reflection/MethodCaller.cs

```diff
@@ -1080,6 +1080,71 @@ public static object CallMethod(object obj, System.Reflection.MethodInfo info)
       }
     }
 
+    /// <summary>
+    /// Throws an exception if a synchronous data portal call is trying to invoke an asynchronous method on the client.
+    /// </summary>
+    /// <param name="isSync">True if the client-side proxy should synchronously invoke the server.</param>
+    /// <param name="obj">Object containing method.</param>
+    /// <param name="methodName">Name of the method.</param>
+    /// <returns></returns>
+    public static void ThrowIfAsyncMethodOnSyncClient(bool isSync, object obj, string methodName)
```

Comment:
This method should probably be part of the data portal, not part of `Csla.Reflection`.